src/waveform/renderers/allshader/waveformrendererstem.cpp

```diff
@@ -5,79 +5,103 @@
 #include <QOpenGLTexture>
 
 #include "engine/engine.h"
+#include "rendergraph/material/rgbamaterial.h"
+#include "rendergraph/vertexupdaters/rgbavertexupdater.h"
 #include "track/track.h"
+#include "util/assert.h"
 #include "util/math.h"
-#include "waveform/renderers/allshader/matrixforwidgetgeometry.h"
-#include "waveform/renderers/allshader/rgbdata.h"
 #include "waveform/renderers/waveformwidgetrenderer.h"
 #include "waveform/waveform.h"
 
+using namespace rendergraph;
+
+namespace {
+constexpr int kMaxSupportedStems = 4;
```

Comment:
This constant is already duplicated twice in `engine/channels/enginedeck.cpp` and `analyzer/constants.h` (my fault). How would you feel about using either of them? analyzer probably makes the most sense here

Reply:
sure, but twice is still one too many :-)

but I don;t see the constant you are referring to in analyser

Reply:
Actually, you are right, I have removed the duplicate in analyser! You should be able to use the one in [here](https://github.com/mixxxdj/mixxx/blob/a54051a30ad9c09381e8659b54fca2d828f79f3b/src/engine/engine.h#L14)

Reply:
Thanks, done!